src/createRestyleFunction.ts

```diff
@@ -73,26 +72,28 @@ const getValue = <Theme extends BaseTheme>(
   return val;
 };
 
-const createRestyleFunction = ({
+const createRestyleFunction = <
+  TProps = Record<string, unknown>,
+  Theme extends BaseTheme = BaseTheme,
+  P extends keyof TProps = keyof TProps,
+  K extends keyof Theme = keyof Theme
+>({
   property,
   transform,
-  styleProperty = property,
+  styleProperty = property.toString(),
```

Comment:
since objects can have numbered or Symbol properties, but the styleProperty type is a string, we need to convert it to a string if we want to assign it like this by default.